tests/infra/server.py

```diff
@@ -75,6 +75,8 @@ def _inject_api_requests_methods(cls):
     def shutdown(self):
         if self._server is not None:
             self._server.server.shutdown()
+        self.join(timeout=10)
```

Comment:
Is 10 seconds always enough? I wonder if this should be made 30 or 60.

Reply:
I'll make it 30.